### DISCUSSION TITLE
What went wrong though I followed all your instructions to use Flux in Forge

### DISCUSSION THREAD

**Discussion by tmprabubiz** [lllyasviel/stable-diffusion-webui-forge#1033] [Q&A]
I HAVE A WINDOWS PC WITH WINDOWS 10 PRO. UPDATED FORGE TO USE FLUX MODELS, FOLLOWING ARE THE STARTUP LINES
==================================================================================
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: f2.0.1v1.10.1-previous-248-gf6ef105c
Commit hash: f6ef105cb333692d8695ebe14c8409f98675e2f4
Launching Web UI with arguments: --forge-ref-a1111-home 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui' --gfpgan-model 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\GFPGAN' --always-normal-vram --always-offload-from-vram --xformers --cuda-malloc --cuda-stream --pin-shared-memory --theme dark --ckpt-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\Stable-diffusion' --vae-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\VAE' --hypernetwork-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\hypernetworks' --embeddings-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\embeddings' --lora-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\lora' --controlnet-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\ControlNet' --controlnet-preprocessor-models-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads'
Using cudaMallocAsync backend.
Total VRAM 8192 MB, total RAM 65447 MB
pytorch version: 2.3.1+cu121
xformers version: 0.0.27
Set vram state to: NORMAL_VRAM
Always offload VRAM
Always pin shared GPU memory
Device: cuda:0 NVIDIA GeForce RTX 2060 SUPER : cudaMallocAsync
VAE dtype preferences: [torch.float32] -> torch.float32
CUDA Using Stream: True
J:\FORGE\system\python\lib\site-packages\transformers\utils\hub.py:127: FutureWarning: Using `TRANSFORMERS_CACHE` is deprecated and will be removed in v5 of Transformers. Use `HF_HOME` instead.
  warnings.warn(
Using xformers cross attention
Using xformers attention for VAE
ControlNet preprocessor location: C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads
[-] ADetailer initialized. version: 24.8.0, num models: 10
sd-webui-prompt-all-in-one background API service started successfully.
2024-08-13 02:57:56,907 - ControlNet - INFO - ControlNet UI callback registered.
Model selected: {'checkpoint_info': {'filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\Stable-diffusion\\Flux\\flux1-dev-fp8.safetensors', 'hash': 'be9881f4'}, 'vae_filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\VAE\\Flux_Vae.safetensors', 'unet_storage_dtype': None}
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 43.5s (prepare environment: 3.4s, launcher: 2.2s, import torch: 4.0s, initialize shared: 0.1s, other imports: 0.9s, load scripts: 15.6s, initialize extra networks: 7.6s, create ui: 5.4s, gradio launch: 3.5s, app_started_callback: 0.7s).
Environment vars changed: {'stream': False, 'inference_memory': 1024.0, 'pin_shared_memory': False}
Model selected: {'checkpoint_info': {'filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\Stable-diffusion\\Flux\\flux1-dev-fp8.safetensors', 'hash': 'be9881f4'}, 'vae_filename': None, 'unet_storage_dtype': None}
Model selected: {'checkpoint_info': {'filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\Stable-diffusion\\Flux\\flux1-dev-fp8.safetensors', 'hash': 'be9881f4'}, 'vae_filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\VAE\\Flux_Vae.safetensors', 'unet_storage_dtype': None}
===================================================================================

SINCE I HAVE NVIDIA GTX 2060SUPER - 8GB GPU, I have tried the following prompt and attributes

PROMPT: Astronaut in a jungle, cold color palette, muted colors, very detailed, sharp focus

Steps: 20, Sampler: Euler, Schedule type: Simple, CFG scale: 1, Distilled CFG Scale: 3.5, Seed: 12345, Size: 896x1152, Model: flux1-dev-fp8, Diffusion with Low Bits: Auto, Swap Method: Queue, Swap Location: CPU, GPU Weights(MB): 7167

But I couldn't create the image in flux
I got the following error message in the command box
==================================================================================
To create a public link, set `share=True` in `launch()`.
Startup time: 43.5s (prepare environment: 3.4s, launcher: 2.2s, import torch: 4.0s, initialize shared: 0.1s, other imports: 0.9s, load scripts: 15.6s, initialize extra networks: 7.6s, create ui: 5.4s, gradio launch: 3.5s, app_started_callback: 0.7s).
Environment vars changed: {'stream': False, 'inference_memory': 1024.0, 'pin_shared_memory': False}
Model selected: {'checkpoint_info': {'filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\Stable-diffusion\\Flux\\flux1-dev-fp8.safetensors', 'hash': 'be9881f4'}, 'vae_filename': None, 'unet_storage_dtype': None}
Model selected: {'checkpoint_info': {'filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\Stable-diffusion\\Flux\\flux1-dev-fp8.safetensors', 'hash': 'be9881f4'}, 'vae_filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\VAE\\Flux_Vae.safetensors', 'unet_storage_dtype': None}
Model selected: {'checkpoint_info': {'filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\Stable-diffusion\\Flux\\flux1-dev-fp8.safetensors', 'hash': 'be9881f4'}, 'vae_filename': None, 'unet_storage_dtype': None}
Loading Model: {'checkpoint_info': {'filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\Stable-diffusion\\Flux\\flux1-dev-fp8.safetensors', 'hash': 'be9881f4'}, 'vae_filename': None, 'unet_storage_dtype': None}
StateDict Keys: {'transformer': 780, 'vae': 244, 'text_encoder': 198, 'text_encoder_2': 220, 'ignore': 0}
Using Detected T5 Data Type: torch.float8_e4m3fn
Using Detected UNet Type: torch.float8_e4m3fn
Working with z of shape (1, 16, 32, 32) = 16384 dimensions.
K-Model Created: {'storage_dtype': torch.float8_e4m3fn, 'computation_dtype': torch.float32}
ac_neg1.png: no embedded information found.
AS-YoungV2-neg.png: no embedded information found.
BadDream.png: no embedded information found.
bad_prompt_version2-neg.png: no embedded information found.
comicbookpencils.png: no embedded information found.
easynegative.png: no embedded information found.
FastNegativeV2.png: no embedded information found.
ImgFixerPre0.3.png: no embedded information found.
NEG-fixl-2.png: no embedded information found.
negativeXL_D.png: no embedded information found.
negative_hand-neg.png: no embedded information found.
ng_deepnegative_v1_75t.png: no embedded information found.
unaestheticXL_bp5.png: no embedded information found.
verybadimagenegative_v1.3.png: no embedded information found.
Model loaded in 8.9s (unload existing model: 0.3s, forge model load: 8.5s).
Skipping unconditional conditioning when CFG = 1. Negative Prompts are ignored.
To load target model ModuleDict
Begin to load 1 model
[Memory Management] Current Free GPU Memory: 7137.20 MB
[Memory Management] Required Model Memory: 5154.62 MB
[Memory Management] Required Inference Memory: 1024.00 MB
[Memory Management] Estimated Remaining GPU Memory: 958.59 MB
Moving model(s) has taken 3.35 seconds
Traceback (most recent call last):
  File "J:\FORGE\webui\modules_forge\main_thread.py", line 37, in loop
    task.work()
  File "J:\FORGE\webui\modules_forge\main_thread.py", line 26, in work
    self.result = self.func(*self.args, **self.kwargs)
  File "J:\FORGE\webui\modules\txt2img.py", line 110, in txt2img_function
    processed = processing.process_images(p)
  File "J:\FORGE\webui\modules\processing.py", line 799, in process_images
    res = process_images_inner(p)
  File "J:\FORGE\webui\modules\processing.py", line 912, in process_images_inner
    p.setup_conds()
  File "J:\FORGE\webui\modules\processing.py", line 1497, in setup_conds
    super().setup_conds()
  File "J:\FORGE\webui\modules\processing.py", line 494, in setup_conds
    self.c = self.get_conds_with_caching(prompt_parser.get_multicond_learned_conditioning, prompts, total_steps, [self.cached_c], self.extra_network_data)
  File "J:\FORGE\webui\modules\processing.py", line 463, in get_conds_with_caching
    cache[1] = function(shared.sd_model, required_prompts, steps, hires_steps, shared.opts.use_old_scheduling)
  File "J:\FORGE\webui\modules\prompt_parser.py", line 262, in get_multicond_learned_conditioning
    learned_conditioning = get_learned_conditioning(model, prompt_flat_list, steps, hires_steps, use_old_scheduling)
  File "J:\FORGE\webui\modules\prompt_parser.py", line 189, in get_learned_conditioning
    conds = model.get_learned_conditioning(texts)
  File "J:\FORGE\system\python\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "J:\FORGE\webui\backend\diffusion_engine\flux.py", line 79, in get_learned_conditioning
    cond_t5 = self.text_processing_engine_t5(prompt)
  File "J:\FORGE\webui\backend\text_processing\t5_engine.py", line 123, in __call__
    z = self.process_tokens([tokens], [multipliers])[0]
  File "J:\FORGE\webui\backend\text_processing\t5_engine.py", line 134, in process_tokens
    z = self.encode_with_transformers(tokens)
  File "J:\FORGE\webui\backend\text_processing\t5_engine.py", line 60, in encode_with_transformers
    z = self.text_encoder(
  File "J:\FORGE\system\python\lib\site-packages\torch\nn\modules\module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "J:\FORGE\system\python\lib\site-packages\torch\nn\modules\module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "J:\FORGE\webui\backend\nn\t5.py", line 205, in forward
    return self.encoder(x, *args, **kwargs)
  File "J:\FORGE\system\python\lib\site-packages\torch\nn\modules\module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "J:\FORGE\system\python\lib\site-packages\torch\nn\modules\module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "J:\FORGE\webui\backend\nn\t5.py", line 186, in forward
    x, past_bias = l(x, mask, past_bias)
  File "J:\FORGE\system\python\lib\site-packages\torch\nn\modules\module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "J:\FORGE\system\python\lib\site-packages\torch\nn\modules\module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "J:\FORGE\webui\backend\nn\t5.py", line 162, in forward
    x, past_bias = self.layer[0](x, mask, past_bias)
  File "J:\FORGE\system\python\lib\site-packages\torch\nn\modules\module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "J:\FORGE\system\python\lib\site-packages\torch\nn\modules\module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "J:\FORGE\webui\backend\nn\t5.py", line 149, in forward
    output, past_bias = self.SelfAttention(self.layer_norm(x), mask=mask, past_bias=past_bias)
  File "J:\FORGE\system\python\lib\site-packages\torch\nn\modules\module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "J:\FORGE\system\python\lib\site-packages\torch\nn\modules\module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "J:\FORGE\webui\backend\nn\t5.py", line 138, in forward
    out = attention_function(q, k * ((k.shape[-1] / self.num_heads) ** 0.5), v, self.num_heads, mask)
  File "J:\FORGE\webui\backend\attention.py", line 314, in attention_xformers
    mask_out[:, :, :mask.shape[-1]] = mask
RuntimeError: The expanded size of the tensor (1) must match the existing size (64) at non-singleton dimension 0.  Target sizes: [1, 256, 256].  Tensor sizes: [64, 256, 256]
The expanded size of the tensor (1) must match the existing size (64) at non-singleton dimension 0.  Target sizes: [1, 256, 256].  Tensor sizes: [64, 256, 256]
*** Error completing request
*** Arguments: ('task(bt9ff001dzkdzvu)', <gradio.route_utils.Request object at 0x00000267E34985B0>, 'Astronaut in a jungle, cold color palette, muted colors, very detailed, sharp focus', '', [], 1, 1, 1, 3.5, 1152, 896, False, 0.7, 2, 'Latent', 0, 0, 0, 'Use same checkpoint', 'Use same sampler', 'Use same scheduler', '', '', None, 0, 20, 'Euler', 'Simple', False, -1, False, -1, 0, 0, 0, False, 'Flux_Vae.safetensors', True, False, False, False, 'base', False, False, {'ad_model': 'face_yolov8n.pt', 'ad_model_classes': '', 'ad_tab_enable': True, 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same VAE', 'ad_use_sampler': False, 'ad_sampler': 'DPM++ 2M', 'ad_scheduler': 'Use same scheduler', 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_use_clip_skip': False, 'ad_clip_skip': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'None', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1, 'is_api': ()}, {'ad_model': 'None', 'ad_model_classes': '', 'ad_tab_enable': True, 'ad_prompt': '', 'ad_negative_prompt': '', 'ad_confidence': 0.3, 'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1, 'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4, 'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked': True, 'ad_inpaint_only_masked_padding': 32, 'ad_use_inpaint_width_height': False, 'ad_inpaint_width': 512, 'ad_inpaint_height': 512, 'ad_use_steps': False, 'ad_steps': 28, 'ad_use_cfg_scale': False, 'ad_cfg_scale': 7, 'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same VAE', 'ad_use_sampler': False, 'ad_sampler': 'DPM++ 2M', 'ad_scheduler': 'Use same scheduler', 'ad_use_noise_multiplier': False, 'ad_noise_multiplier': 1, 'ad_use_clip_skip': False, 'ad_clip_skip': 1, 'ad_restore_face': False, 'ad_controlnet_model': 'None', 'ad_controlnet_module': 'None', 'ad_controlnet_weight': 1, 'ad_controlnet_guidance_start': 0, 'ad_controlnet_guidance_end': 1, 'is_api': ()}, False, 'keyword prompt', 'keyword1, keyword2', 'None', 'textual inversion first', 'Enhance\\add_detail.safetensors', '0.7', 'None', True, False, 1, False, False, False, 1.1, 1.5, 100, 0.7, False, False, True, False, False, 0, 'Gustavosta/MagicPrompt-Stable-Diffusion', '', ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=None, batch_mask_gallery=None, generated_image=None, mask_image=None, mask_image_fg=None, hr_option='Both', enabled=False, module='None', model='None', weight=1, image=None, image_fg=None, resize_mode='Crop and Resize', processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0.0, guidance_end=1.0, pixel_perfect=False, control_mode='Balanced', save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=None, batch_mask_gallery=None, generated_image=None, mask_image=None, mask_image_fg=None, hr_option='Both', enabled=False, module='None', model='None', weight=1, image=None, image_fg=None, resize_mode='Crop and Resize', processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0.0, guidance_end=1.0, pixel_perfect=False, control_mode='Balanced', save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=None, batch_mask_gallery=None, generated_image=None, mask_image=None, mask_image_fg=None, hr_option='Both', enabled=False, module='None', model='None', weight=1, image=None, image_fg=None, resize_mode='Crop and Resize', processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0.0, guidance_end=1.0, pixel_perfect=False, control_mode='Balanced', save_detected_map=True), False, 7, 1, 'Constant', 0, 'Constant', 0, 1, 'enable', 'MEAN', 'AD', 1, False, 1.01, 1.02, 0.99, 0.95, False, 0.5, 2, False, 3, False, 3, 2, 0, 0.35, True, 'bicubic', 'bicubic', False, 0, 'anisotropic', 0, 'reinhard', 100, 0, 'subtract', 0, 0, 'gaussian', 'add', 0, 100, 127, 0, 'hard_clamp', 5, 0, 'None', 'None', False, 'MultiDiffusion', 768, 768, 64, 4, False, False, False, False, False, 'positive', 'comma', 0, False, False, 'start', '', 1, '', '', 0, '', '', 0, '', '', True, False, False, False, False, False, False, 0, False) {}
    Traceback (most recent call last):
      File "J:\FORGE\webui\modules\call_queue.py", line 74, in f
        res = list(func(*args, **kwargs))
    TypeError: 'NoneType' object is not iterable
================================================================================

 I DON'T KNOW, WHAT WENT WRONG. PLEASE GUIDE. I AM VERY EAGER TO USE FLUX IN FORGE.

**Comment by BSG-Walter** [comment]
if you're updating from an older version. try to make a clean install of the new forge
https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/tag/latest

**Reply by tmprabubiz** [reply]
THANK YOU SO MUCH
Your idea worked. I created Flux images in Forge with a clean install
But the time taken for image generation is very high.
Please check the startup Error box 
----------------------
**![forge error](https://github.com/user-attachments/assets/e5bd67a2-8d8b-4f6a-b576-4fe6b1985753)**
----------------------
-Once I click OK, the command prompt continues as below
============================================================
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: f2.0.1v1.10.1-previous-252-g00f1cd36
Commit hash: 00f1cd36bd25431445eb67425ad39434c68c8665
Launching Web UI with arguments: --forge-ref-a1111-home 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui' --gfpgan-model 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\GFPGAN' --always-normal-vram --always-offload-from-vram --xformers --cuda-malloc --cuda-stream --pin-shared-memory --theme dark --ckpt-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\Stable-diffusion' --vae-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\VAE' --hypernetwork-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\hypernetworks' --embeddings-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\embeddings' --lora-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\lora' --controlnet-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\ControlNet' --controlnet-preprocessor-models-dir 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads'
Using cudaMallocAsync backend.
Total VRAM 8192 MB, total RAM 65447 MB
pytorch version: 2.4.0+cu124
WARNING:xformers:WARNING[XFORMERS]: xFormers can't load C++/CUDA extensions. xFormers was built for:
    PyTorch 2.3.1+cu121 with CUDA 1201 (you have 2.4.0+cu124)
    Python  3.10.11 (you have 3.10.6)
  Please reinstall xformers (see https://github.com/facebookresearch/xformers#installing-xformers)
  Memory-efficient attention, SwiGLU, sparse and more won't be available.
  Set XFORMERS_MORE_DETAILS=1 for more details
J:\webui_forge_cu124_torch24\system\python\lib\site-packages\xformers\ops\fmha\flash.py:211: FutureWarning: `torch.library.impl_abstract` was renamed to `torch.library.register_fake`. Please use that instead; we will remove `torch.library.impl_abstract` in a future version of PyTorch.
  @torch.library.impl_abstract("xformers_flash::flash_fwd")
J:\webui_forge_cu124_torch24\system\python\lib\site-packages\xformers\ops\fmha\flash.py:344: FutureWarning: `torch.library.impl_abstract` was renamed to `torch.library.register_fake`. Please use that instead; we will remove `torch.library.impl_abstract` in a future version of PyTorch.
  @torch.library.impl_abstract("xformers_flash::flash_bwd")
J:\webui_forge_cu124_torch24\system\python\lib\site-packages\xformers\ops\swiglu_op.py:128: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  def forward(cls, ctx, x, w1, b1, w2, b2, w3, b3):
J:\webui_forge_cu124_torch24\system\python\lib\site-packages\xformers\ops\swiglu_op.py:149: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  def backward(cls, ctx, dx5):
xformers version: 0.0.27
Set vram state to: NORMAL_VRAM
Always offload VRAM
Always pin shared GPU memory
Device: cuda:0 NVIDIA GeForce RTX 2060 SUPER : cudaMallocAsync
VAE dtype preferences: [torch.float32] -> torch.float32
CUDA Using Stream: True
J:\webui_forge_cu124_torch24\system\python\lib\site-packages\transformers\utils\hub.py:127: FutureWarning: Using `TRANSFORMERS_CACHE` is deprecated and will be removed in v5 of Transformers. Use `HF_HOME` instead.
  warnings.warn(
Using pytorch cross attention
Using pytorch attention for VAE
ControlNet preprocessor location: C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads
J:\webui_forge_cu124_torch24\webui\modules_forge\patch_basic.py:35: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  result = original_loader(*args, **kwargs)
2024-08-13 22:47:02,532 - ControlNet - INFO - ControlNet UI callback registered.
Model selected: {'checkpoint_info': {'filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\Stable-diffusion\\Flux\\flux1DevSchnellBNB_flux1DevBNBNF4.safetensors', 'hash': '0184473b'}, 'additional_modules': [], 'unet_storage_dtype': 'nf4'}
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 348.7s (prepare environment: 2.9s, launcher: 2.2s, import torch: 328.7s, initialize shared: 0.1s, other imports: 0.8s, load scripts: 2.0s, initialize extra networks: 7.8s, create ui: 2.3s, gradio launch: 1.7s).
Environment vars changed: {'stream': False, 'inference_memory': 2187.0, 'pin_shared_memory': False}
Environment vars changed: {'stream': False, 'inference_memory': 1024.0, 'pin_shared_memory': False}
Model selected: {'checkpoint_info': {'filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\Stable-diffusion\\Flux\\flux1DevSchnellBNB_flux1DevBNBNF4.safetensors', 'hash': '0184473b'}, 'additional_modules': [], 'unet_storage_dtype': None}
Environment vars changed: {'stream': False, 'inference_memory': 2187.0, 'pin_shared_memory': False}
Model selected: {'checkpoint_info': {'filename': 'C:\\Users\\REVIVE\\Automatic1111\\stable-diffusion-webui\\models\\Stable-diffusion\\Flux\\flux1DevSchnellBNB_flux1DevBNBNF4.safetensors', 'hash': '0184473b'}, 'additional_modules': [], 'unet_storage_dtype': 'nf4'}
========================================================================
PLEASE GUIDE ME TO PROCEED FURTHER

**Reply by ali0une** [reply]
@tmprabubiz you need to be logged in HF to download some the files like VAE.

VAE is https://huggingface.co/black-forest-labs/FLUX.1-dev/blob/main/ae.safetensors to be putted  in models\VAE

fp16 model is https://huggingface.co/black-forest-labs/FLUX.1-dev/blob/main/flux1-dev.safetensors but you can do without at first, as everything works the same with the fp8 or nf4 models.

Then you'll also need to put in models\text_encoder :
https://huggingface.co/lllyasviel/flux_text_encoders/blob/main/clip_l.safetensors
https://huggingface.co/lllyasviel/flux_text_encoders/blob/main/t5xxl_fp16.safetensors
https://huggingface.co/lllyasviel/flux_text_encoders/blob/main/t5xxl_fp8_e4m3fn.safetensors

You should read again https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/1050 and take a close look at the screenshots.

Hope that helps.

**Comment by webik150** [comment]
Sorry for going off-topic but @tmprabubiz, could you please fix your formatting?

It'll make it much easier to help you if you use the code button to insert logs (or even better three backquotes ```, your log, and then three backquotes again). Especially because some characters in the logs trigger markdown features like headers/titles.
eg:
```
To create a public link, set share=True in launch().
Startup time: 348.7s (prepare environment: 2.9s, launcher: 2.2s, import torch: 328.7s, initialize shared: 0.1s, other imports: 0.8s, load scripts: 2.0s, initialize extra networks: 7.8s, create ui: 2.3s, gradio launch: 1.7s).
Environment vars changed: {'stream': False, 'inference_memory': 2187.0, 'pin_shared_memory': False}
Environment vars changed: {'stream': False, 'inference_memory': 1024.0, 'pin_shared_memory': False}
Model selected: {'checkpoint_info': {'filename': 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\Stable-diffusion\Flux\flux1DevSchnellBNB_flux1DevBNBNF4.safetensors', 'hash': '0184473b'}, 'additional_modules': [], 'unet_storage_dtype': None}
Environment vars changed: {'stream': False, 'inference_memory': 2187.0, 'pin_shared_memory': False}
Model selected: {'checkpoint_info': {'filename': 'C:\Users\REVIVE\Automatic1111\stable-diffusion-webui\models\Stable-diffusion\Flux\flux1DevSchnellBNB_flux1DevBNBNF4.safetensors', 'hash': '0184473b'}, 'additional_modules': [], 'unet_storage_dtype': 'nf4'}
```

this is what it feels like to read issues like this lol

https://github.com/user-attachments/assets/3c3c280a-760e-4ead-9559-55511247c7e4

**Reply by BSG-Walter** [reply]
i'm downloading that
